### DISCUSSION TITLE
Task 6: Productionize 0x API feature with bokeh front end

### DISCUSSION THREAD

**Discussion by icmoore** [defipy-devs/defipy#12] [Ideas]
Productionize the 0x API feature that was implemented for the ETH Denver 2024 BUILD event. The work can be found in the ethdenver branch in the uniswappy package

**Comment by ABusyHippie** [comment]
This is where it is now on CodeSandbox: [Code on CodeSandbox](https://codesandbox.io/p/github/defipy-devs/uniswappy/ethdenver?workspaceId=16e8ef80-7e95-495d-a1e7-1d128dde66f6)

And the (sometimes) live link: [Live Link](https://6pg42n-5006.csb.app/bokeh_server)

Our hackathon submission for context: [Hackathon Submission](https://devfolio.co/projects/defi-paper-trading-live-modeling-tool-with-x-5d78)

**Comment by ABusyHippie** [comment]
Did this last week but forgot to update here. This link [https://dashboard.defipy.org/](https://dashboard.defipy.org/) now points to the ethdenver branch of uniswappy and is live 24/7 (at a small cost of $7/month)

Any changes in that branch will be reflected here so we can continue to make updates that will reflect on the live site to make it more production ready, including easily adding scaling if needed through the service I used to deploy ([Render](https://dashboard.render.com/))

**Comment by ABusyHippie** [comment]
Live link now points to main and I have begun working on .env implementation to protect the premium API key 0x granted us through the JumpStart program. I will use that key for the live link and have added user management via supabase for the ETH Amsterdam hackathon over the last couple days. Then we can allow users to save their own free/premium API key for use with the live link if they prefer, and handle other saved data/preferences/etc with the user management

**Reply by ABusyHippie** [reply]
Note to self, need to add SMTP handler for new Supabase integration in order to have feasible rate limits for emails (currently only 3/hour 😬)

https://supabase.com/docs/guides/platform/going-into-prod#rate-limiting-resource-allocation--abuse-prevention
